### PR TITLE
Compatible toy account

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We wish to add tests that ensure that account details remain as expected.  The recently added API method `get_test_account()` provides useful data, but it is unrefined.  In particular it has no `.did` file entry so the response is hard to parse, and it is hard to compare results with those from `get_account()`.

# Changes
- Make the `get_test_account` response type the same as that of `get_account`.  The type change wraps the response in an opt that is None if the account is not there, and provides the account data in a stable format that doesn't depend on details of the underlying `Account` type.
- Add an entry in the `nns-dapp` `.did` file for `get_test_account`.
  Note: `get_test_account()` has two protections against misuse in production:  It is present only in test builds and it may only be called by a canister controller.

# Tests
- There is a test that verifies that test accounts can be read with the expected format and values.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: rs/backend/src/accounts_store/toy_data.rs [hunk 1]
```diff
@@ -99,15 +98,6 @@ impl AccountsStore {
         index_range_start
     }
```
Comment (bitdivine) [inline rs/backend/src/accounts_store/toy_data.rs:100]: We use the normal `get_account` instead of the special `get_toy_account` method below.  Creating the account identifier is not hard and this way the outputs are consistent.

File: rs/backend/src/main.rs [hunk 2]
```diff
@@ -353,12 +356,16 @@ pub fn create_toy_accounts() {
...
+    over(candid_one, |toy_account_index: u64| {
         let caller = ic_cdk::caller();
         if !ic_cdk::api::is_controller(&caller) {
             dfn_core::api::trap_with("Only the controller may access toy accounts");
         }
-        STATE.with(|s| s.accounts_store.borrow_mut().get_toy_account(toy_account_index as u64))
+        let principal = PrincipalId::new_user_test_id(toy_account_index);
+        STATE.with(|s| match s.accounts_store.borrow().get_account(principal) {
+            Some(account) => GetAccountResponse::Ok(account),
+            None => GetAccountResponse::AccountNotFound,
```
Comment (bitdivine) [inline rs/backend/src/main.rs:367]: To be consistent with `get_account` we wrap the response in an opt if we can find an account, and `None` (or rather `null`) if the requested account is not there.

File: scripts/nns-dapp/downgrade-upgrade-test [hunk 3]
```diff
@@ -48,7 +48,7 @@ get_upgrade_invariant_stats() {
 # Gets some sample accounts
 sample_toy_accounts() {
   echo "Sample accounts will be available once the production commit has a test wasm with get_toy_account()."
-  # seq 0 "$TOY_ACCOUNT_CHUNK_SIZE" "$((NUM_TOY_ACCOUNTS - 1))" | xargs -I{} dfx canister call nns-dapp get_toy_account "({}: nat)" --query
+  # seq 0 "$TOY_ACCOUNT_CHUNK_SIZE" "$((NUM_TOY_ACCOUNTS - 1))" | xargs -I{} dfx canister call nns-dapp get_toy_account "({})" --query
```
Comment (bitdivine) [inline scripts/nns-dapp/downgrade-upgrade-test:51]: This line is not activated yet but we can update it so that it works when we uncomment it.

File: scripts/nns-dapp/test-account.test.data [hunk 4]
```diff
@@ -0,0 +1,93 @@
+
```
Comment (bitdivine) [inline scripts/nns-dapp/test-account.test.data:1]: Look ma: Readable outputs that are consistent with our usual `get_account` responses!

**Comment by dskloetd** [comment]
The PR description has TODO under "Tests" but it looks like some testing is going on. Do you plan to add more tests or did you forget to update the PR description?

**Comment by bitdivine** [comment]
@dskloetd The PR description needed to be updated.  Done.

File: scripts/nns-dapp/test-account.test [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  for ((index = 0; index < NUM_TOY_ACCOUNTS; index++)); do
+    printf "\n===== Account %d =====\n" "$index"
+    dfx canister call nns-dapp get_toy_account "($index)"
+  done
+} >"$actual_output_file"
+
+# Verify that the accounts are as expected
+expected_output="$(cat "$0.data")"
+actual_output="$(cat "$actual_output_file")"
+[[ "$expected_output" == "$actual_output" ]] || {
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply scripts/nns-dapp/test-account.test:54]: Sure, can do.  Done.

**Review by dskloetd — APPROVED** [review]
